File: das_backend/src/main/java/ch/sbb/backend/logging/domain/TenantId.java
package ch.sbb.backend.logging.domain;

import java.util.UUID;

public record TenantId(String tenantId) {

    public TenantId {
        if (tenantId == null || tenantId.isBlank()) {
            throw new IllegalArgumentException("tenantId must not be blank");
        }
        try {
            UUID.fromString(tenantId);
        } catch (IllegalArgumentException e) {
            throw new IllegalArgumentException("Invalid UUID format", e);
        }
    }
}

File: das_backend/src/main/kotlin/ch/sbb/backend/logging/domain/TenantId.kt
package ch.sbb.backend.logging.domain

import java.util.*

@JvmInline
value class TenantId(private val tenantId: String) {

    init {
        require(tenantId.isNotBlank())
        requireNotNull(UUID.fromString(tenantId))
    }
}
